src/wp-includes/rest-api/endpoints/class-wp-rest-pattern-directory-controller.php

```diff
@@ -93,31 +94,19 @@ public function get_items( $request ) {
 		 */
 		require ABSPATH . WPINC . '/version.php';
 
-		$query_args = array(
-			'locale'     => get_user_locale(),
-			'wp-version' => $wp_version,
+		$valid_query_args = array( 'offset', 'order', 'orderby', 'page', 'per_page', 'search', 'slug' );
+		$query_args       = array_merge(
+			array_intersect_key( $request->get_params(), array_flip( $valid_query_args ) ),
+			array(
+				'locale'     => get_user_locale(),
+				'wp-version' => $wp_version, // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable -- it's defined in `version.php` above.
+			)
 		);
```

Comment:
This code can be simplified and made more performant:

```suggestion
		$valid_query_args = array(
			'offset'   => true,
			'order'    => true,
			'orderby'  => true,
			'page'     => true,
			'per_page' => true,
			'search'   => true,
			'slug'     => true,
		);
		$query_args               = array_intersect_key( $request->get_params(), $valid_query_args );
		$query_args['locale']     = get_user_locale();
		$query_args['wp-version'] = $wp_version;
```

The `array_flip()` creates an array with the values of `$valid_query_args`  as the keys. Since this is used only for filtering the args, why not set up the `$valid_query_args` into a keyed structure? In doing so, the `array_flip()` is eliminated.

The `array_merge()` also creates another array. As `'locale'` and `'wp-version'` keys will not be in the array after doing the `array_intersect_key()`, a more performant way is to directly adding them to `$query_args` rather than `array_merge()`.

The changes remove 2 unnecessary array iterations and creations.

Reply:
Fixed in https://github.com/WordPress/wordpress-develop/pull/3861/commits/1cede3c669fdfb0588f2c5562db0fa6c54c10aae.

Reply:
See it in action https://3v4l.org/DGvja

Reply:
Thanks @hellofromtonya !